src/editors/containers/ProblemEditor/data/ReactStateOLXParser.js

```diff
@@ -96,42 +97,91 @@ class ReactStateOLXParser {
     return compoundhint;
   }
 
-  addQuestion() {
+  parseQuestionFromState() {
     const { question } = this.problemState;
-    const questionObject = this.parser.parse(question);
+    const questionObject = new XMLParser({
+        ...this.parserOptions,
+        preserveOrder: true,
+        trimValues: false,
+      }).parse(question);
     return questionObject;
   }
 
+  insertQuestionIntoOLX(OLX, problemType, indentationLevel) {
+    // it's necessary to build the question with the different options, but
+    // this options will mess with the rest of the generated OLX, so we build
+    // them separately, and merge two strings; hacky, but works nicely without
+    // messing up formatting of the question or the rest of the XML
+    const question = this.parseQuestionFromState();
+    const questionBuild = new XMLBuilder({
+      ...this.builderOptions,
+      preserveOrder: true,
+      format: false,
+    }).build(question);
+
+    const tabWidth = 2;
+    const tagPattern = new RegExp(`<${problemType}[^>]*>`);
+
+    const match = OLX.match(tagPattern);
+    const insertIndex = match.index + match[0].length;
+
+    const finalBuild =
+      OLX.slice(0, insertIndex)
+      + `\n${' '.repeat(tabWidth * indentationLevel)}`
```

Comment:
From what I understand, we are adding this bit of complication, just to make the OLX more human-friendly.
Although nice to have, I am debating if its necessary, since with the OLX would never be exposed to any humans after we start using this MFE. Perhaps just adding a newline would suffice ?

Let me know what you think.

Reply:
TLDR is that, imho, either way the code that would solve the problem would be hacky, so I decided in favor of developer experience (and perhaps slightly better performance).

We can avoid having to use regex and merging strings manually either by making the code more complex to accommodate a different structure the builder expects when building with `preserveOrder: true`, or by building the OLX without the question without the option first, then parsing it with `preserveOrder: true`, insert the question object, and then build again. If you think any of these options is better, I'm ok changing it.

Reply:
Actually, my comment was just meant for the `${' '.repeat(tabWidth * indentationLevel)}` portion. :smiley:

Reply:
:+1: removing it.